bpf/process/types/basic.h

```diff
@@ -697,26 +695,41 @@ struct file_buf {
  * a reverse search.
  */
 static inline __attribute__((always_inline)) long
-filter_file_buf(struct file_buf *filter, char *args)
+filter_file_buf(struct file_buf *filter, struct string_buf *args)
 {
 	char *values = (char *)filter + sizeof(struct file_buf); // all values are after the header
-	__u32 remain = filter->total_sz - sizeof(__u32) - sizeof(__u32);
-	int i, next;
+	__u32 next, remain = filter->total_sz - sizeof(__u32) - sizeof(__u32);
+	long i, match;
+
+	/* There are cases where file pointer may not contain a path.
+	 * An example is using an unnamed pipe. This is not a match.
+	 */
+	if (args->len == 0)
```

Comment:
Haven't followed the code closely, but shouldn't we return  for the `not` operators? Maybe `goto: done_filter_file`?

Reply:
In that case, we get an event where the path is an empty string. I don't find much value in having events like this when trying to monitor access to files. This is the reason I kept the same behavior for not operators. 

If this does not make sense, I can change that.

Reply:
OK, that makes sense to me. Thanks!